components/form/ProjectForm.tsx

```diff
@@ -0,0 +1,184 @@
+import useProjectDetail from '@/lib/hooks/useProjectDetail';
+import useProjects from '@/lib/hooks/useProjects';
+import { CustomerDTO, EmployeeDTO, ProjectDTO } from '@/lib/types/dto';
+import { Flex } from '@chakra-ui/layout';
+import { Button, FormControl, FormErrorMessage, FormLabel, Input, Select } from '@chakra-ui/react';
+import { useRouter } from 'next/router';
+import React, { useState } from 'react';
+import CustomerForm from './CustomerForm';
+
+type ProjectFields = Partial<ProjectDTO>;
+
+const validateProjectFields = (form: ProjectFields): ProjectDTO => {
+    const { name, startDate, customer, managingEmployee } = form;
+    if (name && startDate && customer && managingEmployee) {
+        return {
+            ...form,
+            name,
+            startDate,
+            customer,
+            managingEmployee,
+        };
+    } else {
+        throw 'Invalid project form: missing required fields';
+    }
+};
+
+function ProjectFormBody({ project: projectOrNull, customers, employees, onSubmit }: ProjectFormBodyProps) {
+    const router = useRouter();
+    const [projectFields, setProjectFields] = useState<ProjectFields>(projectOrNull || {});
+    const handleCustomerChange = (e: React.FormEvent<HTMLSelectElement>) => {
+        e.preventDefault();
+        const id = e.currentTarget.value;
+        if (id) {
+            const customer = customers.find((el) => el.id === Number(id));
+            setProjectFields({ ...projectFields, customer: customer });
+        }
+    };
+
+    const handleEmployeeChange = (e: React.FormEvent<HTMLSelectElement>) => {
+        e.preventDefault();
+        const id = e.currentTarget.value;
+        if (id) {
+            const employee = employees.find((el) => el.id === Number(id));
+            setProjectFields({ ...projectFields, managingEmployee: employee });
+        }
+    };
+
+    const invalidEndDate =
+        (projectFields.startDate && projectFields.endDate && projectFields.startDate > projectFields.endDate) || false;
```

Comment:
This is altered behavior because `endDate` being `undefined` is a valid case.

Reply:
Hence why it was written as a separate function so there's no need to use nested ternary expressions

Reply:
Can an undefined `endDate` be invalid? If `endDate` is undefined this expression results in false.

Reply:
The logic is that endDate doesn't need to be defined, but we should always know at least when project or anything else has started
https://github.com/three-consulting/epoc-backend/blob/main/src/main/kotlin/three/consulting/epoc/dto/ProjectDTO.kt#L13-L14
The service layer checks if `endDate` is not null, that it's not before `startDate`. So right now null `endDate` is perfectly fine.

Reply:
So `endDate` is `null`, here we get `invalidEndDate === false` as intended?

Reply:
AAAAAAAA it's INvalid. I don't know how to read. I read `isValidEndDate`